homeassistant/components/tesla/binary_sensor.py

```diff
@@ -15,6 +15,21 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
         for device in hass.data[TESLA_DOMAIN]["devices"]["binary_sensor"]
     ]
     add_entities(devices, True)
+    return True
+
+
+async def async_setup_entry(hass, config_entry, async_add_devices):
+    """Set up the Tesla binary_sensors by config_entry."""
+    return await hass.async_add_executor_job(
+        setup_platform, hass, config_entry.data, async_add_devices, None
+    )
+
+
+async def async_unload_entry(hass, entry) -> bool:
```

Comment:
Platforms don't support this api. We forward the unloading to the platform component which unloads its platforms via the platform helper. Entities will then be removed and we can use `async_will_remove_from_hass` entity coroutine method to unsubscribe entity specific listeners.

Central listeners need to be unsubscribed from the tesla component `async_unload_entry` function.

Reply:
I don't understand what you want done. You want me to delete `async_unload_entry` in any platform and leave it only in `__init__`?